contracts/colony/ColonyExpenditure.sol

```diff
@@ -174,6 +175,43 @@ contract ColonyExpenditure is ColonyStorage {
     expenditureSlots[_id][_slot].claimDelay = _claimDelay;
   }
 
+  uint256 constant EXPENDITURES_SLOT = 25;
+  uint256 constant EXPENDITURESLOTS_SLOT = 26;
+  uint256 constant EXPENDITURESLOTPAYOUTS_SLOT = 27;
+
+  function setExpenditureState(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _id,
+    uint256 _slot,
+    bool[] memory _mask,
+    bytes32[] memory _keys,
+    bytes32 _value
+  )
+    public
+    stoppable
+    authDomain(_permissionDomainId, _childSkillIndex, expenditures[_id].domainId)
+  {
+    require(_keys.length > 0, "colony-expenditure-no-keys");
+
+    require(
+      _slot == EXPENDITURES_SLOT ||
+      _slot == EXPENDITURESLOTS_SLOT ||
+      _slot == EXPENDITURESLOTPAYOUTS_SLOT,
+      "colony-expenditure-bad-slot"
+    );
+
+    // Only allow editing expenditure status, owner, and finalizedTimestamp
+    //  Note that status + owner occupy one slot
+    if (_slot == EXPENDITURES_SLOT) {
+      uint256 offset = uint256(_keys[0]);
+      require(_keys.length == 1, "colony-expenditure-bad-keys");
```

Comment:
Because this is the only case where we need to restrict the inputs to avoid letting users edit protected state. For the other slots everything is on the table, and while users could in principle pass very long arrays of gibberish, it'll end up pointing to random storage slots so it's not really an issue.

Reply:
Although I suppose I could do

```
uint256 offset = uint256(_keys[_keys.length - 1]);
```

And skip the require.